index.html

```diff
@@ -1969,39 +2017,56 @@ <h2>The <dfn>NDEFRecordType</dfn> string</h2>
     </p>
   </section> <!-- visibility & focus: the suspended state -->
 
-  <section><h3>Aborting pending push operation</h3>
+  <section><h3>Clearing the pending push tuple</h3>
   <p>
-    To attempt to <dfn>abort a pending push operation</dfn> on an
+    To <dfn>clear the pending push tuple</dfn> on an
     <a>environment settings object</a>, perform the following steps:
     <ol class=algorithm>
       <li>
         If there is no <a>pending push tuple</a> |tuple|, abort these steps.
       </li>
       <li>
-        Let |writer:NFCWriter| be |tuple|'s writer.
+        Let |writer| be the <a>pending push tuple</a>'s writer.
+      </li>
+      <li>
+        Stop |writer|.[[\PushTimer]] if active.
+      </li>
+      <li>
+        Stop |writer|.[[\PushHandler]] if running.
+      </li>
+      <li>
+        Clear the <a>pending push tuple</a>.
+      </li>
+    </ol>
+  </p>
+</section>
+
+  <section><h3>Aborting pending push operation</h3>
+  <p>
+    To attempt to <dfn>abort a pending push operation</dfn> on an
+    <a>environment settings object</a>, perform the following steps:
+    <ol class=algorithm>
+      <li>
+        If there is no <a>pending push tuple</a> |tuple|, abort these steps.
       </li>
       <li>
         Let |p:Promise| be |tuple|'s promise.
       </li>
       <li>
-        Stop the |writer|'s |timer| if it is active.
+        Let |writer| be the <a>pending push tuple</a>'s writer.
       </li>
       <li>
-        If the |writer| has already initiated NFC data transfer,
-        abort these steps.
+        If |writer|.[[\PushHandler]] has already initiated an ongoing NFC data
+        transfer, abort these steps.
       </li>
       <li>
-        Clear the <a>pending push tuple</a>.
+        Run the <a>clear the pending push tuple</a> steps.
```

Comment:
Don't we have clean up steps associated the settlement of the promise? We reject it below so maybe this is unneeded

Reply:
AFAIK we don't have implicit cleanup steps after settling a promise, at least in this spec. Are they defined elsewhere? (that might explain the now deleted note about cleanup on promise rejection).

Reply:
That is what the [=promise/React=] does. It adds steps to be run whenever a promise resolves or rejects.

We used to use a transformed promise for that, but now WebIDL handles this

Reply:
It is much simpler